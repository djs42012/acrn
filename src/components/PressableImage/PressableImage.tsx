import React from 'react';
import { ImageProps, Image, Pressable, PressableProps } from 'react-native';

type PressableWithoutStyleProps = Omit<PressableProps, 'style'>;
interface ProfileProps extends ImageProps, PressableWithoutStyleProps {
  height: number;
  width: number;
  callback?: () => any;
  imageFlex?: number;
  imageBackgroundColor?: string;
  imageBorderRadius?: number;
  imageBorderColor?: string;
  imagePadding?: number;
  imageMargin?: number;
  imageBorderWidth?: number;
}

function PressableImage(props: ProfileProps) {
  const {
    accessibilityLabel,
    accessibilityHint,
    source,
    height,
    width,
    callback,
    imageFlex,
    imageBackgroundColor,
    imageBorderRadius,
    imageBorderColor,
    imagePadding,
    imageMargin,
    imageBorderWidth,
  } = props;
  return (
    <Pressable
      onPress={callback}
      accessibilityLabel={accessibilityLabel}
      accessibilityHint={accessibilityHint}
      style={{
        flex: imageFlex,
        backgroundColor: imageBackgroundColor,
        borderRadius: imageBorderRadius,
        borderColor: imageBorderColor,
        padding: imagePadding,
        margin: imageMargin,
        borderWidth: imageBorderWidth,
      }}
    >
      <Image source={source} style={{ width, height }} />
    </Pressable>
  );
}

export default PressableImage;
