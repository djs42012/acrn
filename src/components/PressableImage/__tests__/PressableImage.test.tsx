import React from 'react';
import PressableImage from '../PressableImage';
import { Image } from 'react-native';
import TestRenderer from 'react-test-renderer';

describe('PressableImage test', () => {
  let profile: any;
  let profileRoot: any;
  const props = {
    height: 100,
    width: 100,
  };
  beforeEach(() => {
    profile = TestRenderer.create(<PressableImage source={0} {...props} />);
    profileRoot = profile.root;
    expect(profile.toJSON()).toMatchSnapshot();
  });
  test('style testing', () => {
    expect(profileRoot.findByType(Image).props.style).toMatchObject({
      height: 100,
      width: 100,
    });
  });
});
