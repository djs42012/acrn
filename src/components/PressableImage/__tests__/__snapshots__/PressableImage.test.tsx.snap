// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PressableImage test placeholder 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
>
  <Image
    source={0}
    style={
      Object {
        "height": 100,
        "width": 100,
      }
    }
  />
</View>
`;

exports[`PressableImage test style testing 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": undefined,
      "borderColor": undefined,
      "borderRadius": undefined,
      "borderWidth": undefined,
      "flex": undefined,
      "margin": undefined,
      "padding": undefined,
    }
  }
>
  <Image
    source={0}
    style={
      Object {
        "height": 100,
        "width": 100,
      }
    }
  />
</View>
`;
